vms/evm/database/wrapped_database.go

```diff
@@ -0,0 +1,61 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package database
+
+import (
+	"errors"
+
+	"github.com/ava-labs/libevm/ethdb"
+
+	"github.com/ava-labs/avalanchego/database"
+)
+
+var (
+	_ ethdb.Batch         = (*ethBatchWrapper)(nil)
+	_ ethdb.KeyValueStore = (*ethDBWrapper)(nil)
+
+	ErrSnapshotNotSupported = errors.New("snapshot is not supported")
+	ErrStatNotSupported     = errors.New("stat is not supported")
+)
+
+func WrapDatabase(db database.Database) ethdb.KeyValueStore { return ethDBWrapper{db} }
+
+type ethDBWrapper struct{ database.Database }
+
+func (ethDBWrapper) Stat(string) (string, error) { return "", ErrStatNotSupported }
+
+func (db ethDBWrapper) NewBatch() ethdb.Batch { return ethBatchWrapper{db.Database.NewBatch()} }
+
+// TODO: propagate size through avalanchego Database interface
+func (db ethDBWrapper) NewBatchWithSize(int) ethdb.Batch {
+	return ethBatchWrapper{db.Database.NewBatch()}
+}
+
+func (ethDBWrapper) NewSnapshot() (ethdb.Snapshot, error) {
+	return nil, ErrSnapshotNotSupported
+}
+
+// This method assumes that the prefix is NOT part of the start, so there's
+// no need for the caller to prepend the prefix to the start.
```

Comment:
This comment just seems weird to me - the function signature is literally:
```
NewIterator(prefix []byte, start []byte)
```

The prefix and start are two distinct parameters, so I'm not sure why anyone would ever assume that the start should have the prefix prepended to it. I feel like we could just get rid of this godoc.